pages/advanced-algorithms/available-algorithms/leiden_community_detection.mdx

```diff
@@ -0,0 +1,160 @@
+---
+
+import { Steps } from 'nextra/components'
+import { Callout } from 'nextra/components'
+import { Card, Cards } from 'nextra/components'
+import GitHub from '/components/icons/GitHub'
+
+# leiden_community_detection
+
+Community in graphs mirrors real-world communities, like social circles. In a
+graph, communities are sets of nodes. M. Girvan and M. E. J. Newman note that
+nodes in a community connect more intensely with each other than with outside
+nodes.
+
+This module employs the [Leiden
+algorithm](https://en.wikipedia.org/wiki/Leiden_algorithm) for community detection
+based on paper [*From Louvain to Leiden: guaranteeing well-connected communities*](https://arxiv.org/abs/1810.08473). 
+The Leiden algorithm is a hierarchical clustering algorithm, that recursively merges communities into single nodes by greedily optimizing the modularity and the process repeats in the condensed graph.
+It enhances the Louvain algorithm by addressing its limitations, particularly in situations where some identified communities are not well-connected. 
+This improvement is made by periodically subdividing communities into smaller, well-connected groups.
+With an $\mathcal{O}(Lm)$ runtime for $m$ edges and $L$ number of iterations, it suits large graphs.
+
+<Cards>
+  <Card
+    icon={<GitHub />}
+    title="Source code"
+    href="https://github.com/memgraph/mage/blob/main/cpp/leiden_community_detection_module/leiden_community_detection_module.cpp"
+  />
+</Cards>
+
+| Trait                    | Value                                                                                                       |
+| ------------------------ | --------------------- |
+| **Module type**          | algorithm             |
+| **Implementation**       | C++                   |
+| **Graph direction**      | undirected            |
+| **Relationship weights** | weighted / unweighted |
+| **Parallelism**          | parallel              |
+
+## Procedures
+
+<Callout type="info">
+You can execute this algorithm on [graph projections, subgraphs or portions of the graph](/advanced-algorithms/run-algorithms#run-procedures-on-subgraph).
+</Callout>
+
+### `get()`
+
+Computes graph communities using the Leiden algorithm.
+
+{<h4> Input: </h4>}
+
+- `subgraph: Graph` (**OPTIONAL**) ➡ A specific subgraph, which is an [object of type Graph](/advanced-algorithms/run-algorithms#run-procedures-on-subgraph) returned by the `project()` function, on which the algorithm is run.
+- `weight: string (default=null)` ➡ Specifies the default relationship weight. If not set, 
+  the algorithm uses the `weight` relationship attribute when present and otherwise 
+  treats the graph as unweighted.
+- `gamma: double (default=1.0)` ➡ Resolution parameter used when computing the modularity. Internally the value is divided by the number of relationships for an unweighted graph, or the sum of weights of all relationships otherwise.
```

Comment:
Why is gamma divided by some kind of weight? Please explain.

Reply:
This statement isn’t entirely accurate. Gamma isn’t divided on its own; instead, it’s divided by the specified value in the modularity optimization formula. I don’t think additional explanation is necessary since we link to the paper in the docs. Going into detail on individual parts of the algorithm would be better suited for a blog post than the docs, so I would just delete that part of the sentence.
![Screenshot from 2024-10-25 11-56-05](https://github.com/user-attachments/assets/00b02a2d-7737-4c35-9cbb-25b650c4b4e4)

Reply:
In that case, please mention that the paper has the details.